File: registration-service/registration-service/src/main/java/vn/edu/crs/registration_service/entity/Registration.java
package vn.edu.crs.registration_service.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "registrations")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Registration {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long studentId;

    @Column(nullable = false)
    private Long courseId;

    @Column(nullable = false)
    private LocalDateTime ngayDangKy;

    @Column(nullable = false, length = 20)
    private String trangThai; // "DA_DANG_KY", "DA_HUY"
}

File: crs-frontend/src/types/registration.ts
export interface RegistrationRequest {
    studentId: number;
    courseId: number;
}

export interface Registration {
    id: number;
    studentId: number;
    courseId: number;
    ngayDangKy: string;
    trangThai: 'DA_DANG_KY' | 'DA_HUY';
}